dotnet/Dockerfile

```diff
@@ -1,15 +1,20 @@
 FROM gitpod/workspace-full:latest
 
 USER root
-RUN wget -q https://packages.microsoft.com/config/ubuntu/19.04/packages-microsoft-prod.deb && \
+
+RUN wget -q "https://packages.microsoft.com/config/ubuntu/19.04/packages-microsoft-prod.deb" && \
     dpkg -i packages-microsoft-prod.deb && \
     rm -rf packages-microsoft-prod.deb && \
     add-apt-repository universe && \
     apt-get update && \
-    echo resolvconf resolvconf/linkify-resolvconf boolean false | debconf-set-selections && \
-    DEBIAN_FRONTEND=noninteractive ACCEPT_EULA=Y apt-get -y -o APT::Install-Suggests="true" install \
+    DEBIAN_FRONTEND=noninteractive ACCEPT_EULA=Y apt-get --no-install-recommends -y -o APT::Install-Suggests="true" install \
         dotnet-sdk-2.2 \
         dotnet-sdk-3.1 \
+    && apt-key adv --keyserver "hkp://keyserver.ubuntu.com:80" --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF \
+    && echo "deb https://download.mono-project.com/repo/ubuntu stable-bionic main" | tee /etc/apt/sources.list.d/mono-official-stable.list \
+    && apt-get update \ 
+    && apt-get install -yq \
```

Comment:
It feels a bit strange to me to do `apt-get update` and `apt-get install` twice in a row.

Do you think it's possible to do all the `add-apt-repository` and `apt-key adv` and `tee /etc/apt/sources.list` _before_, and then run `apt-get update` once, and then `apt-get install` all the things in one go?

Reply:
Also I think it's better to not use `USER root`, as that encourages others to do dangerous things that could prevent them from being able to access their own files.

Better to use `USER gitpod` (which is the default) and then use `sudo` everywhere it's needed.

Reply:
I agree with Jan, permission issues are a huge pain to deal with.

Reply:
Done! And Stable!